tango_simlib/tango_launcher.py

```diff
@@ -30,6 +30,8 @@
                   "same number of times and the names and classes are matched in order")
 required_argument('--server-command', help="TANGO server executable command")
 required_argument('--server-instance', help="TANGO server instance name")
+required_argument('--file', help="ASCII file containing device configuration parameters" +
+                  "and device network access parameter")
```

Comment:
I suggest this should be an option (running simulators without Tango DB) and not enforced as it may not always be the desired use case. 
Also need to think about how this will play with the existing "--put-device-property" argument.

Reply:
Agree, it should be optional.  Would also like the description string to say that this is if you are NOT using a TANGO DB server.

Reply:
def generate_db_file(self, server, instance, device,
                         tangoclass=None, properties={}):
        """Generate a database file corresponding to the given arguments."""
        if not tangoclass:
            tangoclass = server
        # Open the file
        with open(self.db, 'w') as f:
            f.write("/".join((server, instance, "DEVICE", tangoclass)))
            f.write(': "' + device + '"\n')
        # Create database
        db = Database(self.db)
        # Patch the property dict to avoid a PyTango bug
        patched = dict((key, value if value != '' else ' ')
                       for key, value in properties.items())
        # Write properties
        db.put_device_property(device, patched)
        return db

Reply:
That's how the `tango.test_context` module handle's device property registration when using the file as a database.